src/vs/workbench/contrib/search/browser/searchActions.ts

```diff
@@ -431,30 +431,20 @@ class ReplaceActionRunner {
 		@IReplaceService private readonly replaceService: IReplaceService,
 		@IEditorService private readonly editorService: IEditorService,
 		@IConfigurationService private readonly configurationService: IConfigurationService,
-		@IUriIdentityService private readonly uriIdentityService: IUriIdentityService
+		@IUriIdentityService private readonly uriIdentityService: IUriIdentityService,
+		@IViewsService private readonly viewsService: IViewsService
 	) { }
 
 	async performReplace(element: FolderMatch | FileMatch | Match): Promise<any> {
 		// since multiple elements can be selected, we need to check the type of the FolderMatch/FileMatch/Match before we perform the replace.
 		const opInfo = getElementsToOperateOnInfo(this.viewer, element, this.configurationService.getValue<ISearchConfigurationProperties>('search'));
 		const elementsToReplace = opInfo.elements;
 
-		await Promise.all(elementsToReplace.map(async (elem) => {
-			const parent = elem.parent();
-
-			if ((parent instanceof FolderMatch || parent instanceof FileMatch) && arrayContainsElementOrParent(parent, elementsToReplace)) {
-				// skip any children who have parents in the array
-				return;
-			}
+		const searchResult = getSearchView(this.viewsService)?.searchResult;
```

Comment:
This way of getting the `SearchResult` seems awkward. It feels better to get it from the model itself, like walking up the `.parent` chain on the elements. But this is fine for now, can think about it some more...

Reply:
I was thinking that it would be a little messy to deal with the case where the search model is ill-formed and then I can't find a search model from walking up the parent chain of one of the matches.